i2c/bmp280_i2c/bmp280_i2c.c

```diff
@@ -0,0 +1,253 @@
+
+    GPIO PICO_DEFAULT_I2C_SDA_PIN (on Pico this is GP4 (pin 6)) -> SDA on BMP280
+    board
+    GPIO PICO_DEFAULT_I2C_SCK_PIN (on Pico this is GP5 (pin 7)) -> SCL on
+    BMP280 board
+    3.3v (pin 36) -> VCC on BMP280 board
+    GND (pin 38)  -> GND on BMP280 board
+ */
+
+ // device has default bus address of 0x76
+#define ADDR _u(0x76)
+
+// hardware registers
+#define REG_CONFIG _u(0xF5)
+#define REG_CTRL_MEAS _u(0xF4)
+#define REG_RESET _u(0xE0)
+
+#define REG_TEMP_XLSB _u(0xFC)
+#define REG_TEMP_LSB _u(0xFB)
+#define REG_TEMP_MSB _u(0xFA)
+
+#define REG_PRESSURE_XLSB _u(0xF9)
+#define REG_PRESSURE_LSB _u(0xF8)
+#define REG_PRESSURE_MSB _u(0xF7)
+
+// calibration registers
+#define REG_DIG_T1_LSB _u(0x88)
+#define REG_DIG_T1_MSB _u(0x89)
+#define REG_DIG_T2_LSB _u(0x8A)
+#define REG_DIG_T2_MSB _u(0x8B)
+#define REG_DIG_T3_LSB _u(0x8C)
+#define REG_DIG_T3_MSB _u(0x8D)
+#define REG_DIG_P1_LSB _u(0x8E)
+#define REG_DIG_P1_MSB _u(0x8F)
+#define REG_DIG_P2_LSB _u(0x90)
+#define REG_DIG_P2_MSB _u(0x91)
+#define REG_DIG_P3_LSB _u(0x92)
+#define REG_DIG_P3_MSB _u(0x93)
+#define REG_DIG_P4_LSB _u(0x94)
+#define REG_DIG_P4_MSB _u(0x95)
+#define REG_DIG_P5_LSB _u(0x96)
+#define REG_DIG_P5_MSB _u(0x97)
+#define REG_DIG_P6_LSB _u(0x98)
+#define REG_DIG_P6_MSB _u(0x99)
+#define REG_DIG_P7_LSB _u(0x9A)
+#define REG_DIG_P7_MSB _u(0x9B)
+#define REG_DIG_P8_LSB _u(0x9C)
+#define REG_DIG_P8_MSB _u(0x9D)
+#define REG_DIG_P9_LSB _u(0x9E)
+#define REG_DIG_P9_MSB _u(0x9F)
+
+// number of calibration registers to be read
+#define NUM_CALIB_PARAMS 24
+
+struct bmp280_calib_param {
+    // temperature params
+    uint16_t dig_t1;
+    int16_t dig_t2;
+    int16_t dig_t3;
+
+    // pressure params
+    uint16_t dig_p1;
+    int16_t dig_p2;
+    int16_t dig_p3;
+    int16_t dig_p4;
+    int16_t dig_p5;
+    int16_t dig_p6;
+    int16_t dig_p7;
+    int16_t dig_p8;
+    int16_t dig_p9;
+};
+
+#ifdef i2c_default
+void bmp280_init() {
+    // use the "handheld device dynamic" optimal setting (see datasheet)
+    uint8_t buf[2];
+
+    // 500ms sampling time, x16 filter, not set
+    const uint8_t reg_config_val = ((0x04 << 5) | (0x05 << 2)) & 0xFC;
+
+    // LSB of slave address sets read or write mode
+    // send register number followed by its corresponding value
+    buf[0] = REG_CONFIG;
+    buf[1] = reg_config_val;
+    i2c_write_blocking(i2c_default, ADDR, buf, 2, false);
+
+    // osrs_t x1, osrs_p x4, normal mode operation
+    const uint8_t reg_ctrl_meas_val = (0x01 << 5) | (0x03 << 2) | (0x03);
+    buf[0] = REG_CTRL_MEAS;
+    buf[1] = reg_ctrl_meas_val;
+    i2c_write_blocking(i2c_default, ADDR, buf, 2, false);
+}
+
+void bmp280_read_raw(int32_t* temp, int32_t* pressure) {
+    // BMP280 data registers are auto-incrementing and we have 3 temperature and
+    // pressure registers each, so we start at 0xF7 and read 6 bytes to 0xFC note:
+    // normal mode does not require further ctrl_meas and config register writes
```

Comment:
Perhaps the `note:` should be moved from the end of the line to the start of the next line?

Reply:
Agreed!